src/sssom/parsers.py

```diff
@@ -425,7 +425,7 @@ def from_sssom_dataframe(
     # This is to address: A value is trying to be set on a copy of a slice from a DataFrame
     if CONFIDENCE in df.columns:
         df2 = df.copy()
-        df2[CONFIDENCE].replace(r"^\s*$", np.nan, regex=True, inplace=True)
+        df2[CONFIDENCE] = df2[CONFIDENCE].replace(r"^\s*$", np.nan, regex=True)
```

Comment:
LGTM. These two lines should be equivalent.

Reply:
Same, LGTM.

But looking at the larger context, I don’t understand the point of the whole `df2 = df.copy() ... df = df2` dance.

The whole block could simply be:

```python
if CONFIDENCE in df.columns:
    df.replace({CONFIDENCE: r"^\s*$"}, np.nan, regex=True, inplace=True)
```

Reply:
Thank you both, @gouttegd I have adopted your suggestion, thanks!